esports_data_from_rimble/main.go

```diff
@@ -74,42 +64,29 @@ type SecretArgs struct {
 	RimbleAPIKey string `json:"api_key"`
 }
 
-type MatchWinnerArgs struct {
-	Date    string `json:"date"`
-	MatchID string `json:"match_id"`
-}
-
 //export matchWinnerFromRimble
-func matchWinnerFromRimble(inputPtr uint64, secretPtr uint64) uint64 {
-	var input MatchWinnerArgs
-	inputData := basm.ReadFromHost(inputPtr)
-	err := json.Unmarshal(inputData, &input)
-	if err != nil {
-		outErr := fmt.Errorf("could not unmarshal input args: %w", err)
-		return WriteError(outErr)
-	}
-
+func matchWinnerFromRimble(_ uint64, secretPtr uint64) uint64 {
 	var secret SecretArgs
 	secretData := basm.ReadFromHost(secretPtr)
-	err = json.Unmarshal(secretData, &secret)
+	err := json.Unmarshal(secretData, &secret)
 	if err != nil {
 		outErr := fmt.Errorf("could not unmarshal secret args: %w", err)
 		return WriteError(outErr)
 	}
 
-	match, err := getMatchDataFromRimble(
-		input.Date,
-		input.MatchID,
-		secret.RimbleAPIKey,
-	)
-	if err != nil {
-		outErr := fmt.Errorf("getting match data: %w", err)
+	matches, err := getRecentMatchesFromRimble(secret.RimbleAPIKey)
+	switch {
+	case err != nil:
+		outErr := fmt.Errorf("getting recent matches: %w", err)
+		return WriteError(outErr)
+	case len(matches) == 0:
+		outErr := fmt.Errorf("no recent matches found")
 		return WriteError(outErr)
 	}
 
-	matchWinner, err := TeamWinner(match, input.Date)
+	matchWinner, err := TeamWinner(matches[0])
```

Comment:
both the "match winner" here and "team kill diff" logic further down behave the same except that we are using the first most recent match returned by Rimble as opposed to fetching one with a hardcoded match id